### PR TITLE
Adding compliance migration for comp run info

### PR DESCRIPTION
**Pull request by YashviJain01**
Signed-off-by: Yashvi Jain <yashvi.jain@progress.com>

### :nut_and_bolt: Description: What code changed, and why?

Now a new index with comp-3-run-info will be generated on starting the service with new mapping and older data

### :+1: Definition of Done

Comp-3-run-info created and deleted order index

### :athletic_shoe: How to Build and Test the Change

```
rebuild components/compliance-service/

```

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/compliance-service/reporting/relaxting/migrations.go [hunk 1]
```diff
@@ -239,11 +240,22 @@ func RunMigrations(backend ES2Backend, statusSrv *statusserver.Server) error {
 	a2V6Indices := A2V6ElasticSearchIndices{backend: &backend}
 	err = backend.migrate(a2V6Indices, statusSrv, statusserver.MigrationLabelESa2v6)
 	if err != nil {
-		errMsg := errors.Wrap(err, fmt.Sprintf("%s, migration failed for %s", myName, statusserver.MigrationLabelESa2v5))
+		errMsg := errors.Wrap(err, fmt.Sprintf("%s, migration failed for %s", statusserver.MigrationLabelESa2v5))
```
Comment (Dmaddu) [inline components/compliance-service/reporting/relaxting/migrations.go:243]: here fmt.Sprintf expecting two values, but you are sending one value

File: components/compliance-service/reporting/relaxting/migrations_from_a2_v7.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	olderIndexCompRunInfo      = a2V7CompRunInfoIndexPrefix + "run-info"
+)
+
+type A2V7ElasticSearchIndices struct {
+	backend *ES2Backend
+}
+
+//migrateCompRunInfo migrates the comp run info table
+func (migratable A2V7ElasticSearchIndices) migrateCompRunInfo() error {
+	myName := "A2V1ElasticSearchIndices::migrateCompRunInfo"
```
Comment (Dmaddu) [inline components/compliance-service/reporting/relaxting/migrations_from_a2_v7.go:21]: A2V7

File: components/compliance-service/reporting/relaxting/migrations_from_a1.go [hunk 3]
```diff
@@ -20,6 +20,10 @@ type A1ElasticSearchIndices struct {
 	backend *ES2Backend
 }
 
+func (migratable A1ElasticSearchIndices) migrateCompRunInfo() error {
```
Comment (Dmaddu) [inline components/compliance-service/reporting/relaxting/migrations_from_a1.go:23]: we should not touch these migrations, these are for time series migrations, we might need to write single migration file

**Review by Dmaddu — CHANGES_REQUESTED** [review]

File: components/compliance-service/reporting/relaxting/migrations_from_a2_v2_comp_run.go [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+	return nil
+}
+
+func (a A2V2CompRunIndices) postFeedsMigration() error {
+	return nil
+}
+
+//removes the older index thats not needed
+func (migratable A2V2CompRunIndices) postMigration() error {
+	myName := "A2V7ElasticSearchIndices::postMigration"
```
Comment (Dmaddu) [inline components/compliance-service/reporting/relaxting/migrations_from_a2_v2_comp_run.go:61]: A2V2CompRunIndices

**Review by Dmaddu — CHANGES_REQUESTED** [review]

**Review by Dmaddu — APPROVED** [review]

**Review by kalroy — APPROVED** [review]

**Pull request merged**